### PR TITLE
Remove version specific prefixes from builders

### LINKED ISSUE #3085: Remove version-specific prefixes from builder methods
With the split of the builders into per-version kinds (see #2842 and #2851), I feel there's a new question around whether we should remove the prefixes from the method names. They used to exist because the builders were combined over multiple HTTP versions, so the options were scoped to a version. But now:

```rust
mod http1 {
    impl Builder {
        pub fn http1_title_case_headers() {
            // ...
        }
    }
}
```

It seems all those can have the `http1_` prefix removed, and same for the `http2::Builder`s.

Noticed in https://github.com/hyperium/hyper-util/pull/11#discussion_r1035315089.

### PR DESCRIPTION
**Pull request by Michael-J-Ward**
Fixes #3085 

Should I also change thet non-public methods in `proto::h1::conn`?
  - `set_http1_header_read_timeout`
  - `set_h1_parser_config`

### CODE REVIEW

File: src/server/conn/http1.rs [hunk 1]
```diff
@@ -261,7 +261,7 @@ impl Builder {
     /// Note that this setting does not affect HTTP/2.
```
Comment (oliviacrain) [inline src/server/conn/http1.rs:261]: There are a few comments with this note- can we remove these as well, since they no longer apply?

Comment (Michael-J-Ward) [reply src/server/conn/http1.rs]: Good catch, done.

**Review by seanmonstar — APPROVED** [review]
This looks super, thanks so much! I see there's a `fixup!` commit in the list, did you want to rebase and do the fix locally? The merge button only lets me either rebase merge without interactive mode, or squash it all into a single commit.

**Comment by Michael-J-Ward** [comment]
done ✅

**Pull request merged**